crates/biome_cli/src/execute/migrate/eslint_to_biome.rs

```diff
@@ -24,8 +27,171 @@ pub(crate) struct MigrationOptions {
 
 #[derive(Debug, Default)]
 pub(crate) struct MigrationResults {
+    /// Path to the migrated ESlint configuration
+    pub(crate) eslint_path: Option<Box<str>>,
+    /// Is the Biome configuration updated?
+    pub(crate) write: bool,
     // Contains inspired rules that were not migrated because `include_inspired` is disabled
-    pub(crate) has_inspired_rules: bool,
+    pub(crate) inspired: BTreeSet<EslintRuleName>,
+    pub(crate) nursery: BTreeSet<EslintRuleName>,
+    pub(crate) migrated: BTreeSet<EslintRuleName>,
+    pub(crate) unsupported: BTreeSet<EslintRuleName>,
+}
+impl MigrationResults {
+    pub(crate) fn add(&mut self, sourced_rule: &str, status: RuleMigrationResult) {
+        let sourced = EslintRuleName::from_str(sourced_rule);
+        match status {
+            RuleMigrationResult::Migrated => {
+                self.migrated.insert(sourced);
+            }
+            RuleMigrationResult::Inspired => {
+                self.inspired.insert(sourced);
+            }
+            RuleMigrationResult::Nursery => {
+                self.nursery.insert(sourced);
+            }
+            RuleMigrationResult::Unsupported => {
+                self.unsupported.insert(sourced);
+            }
+        }
+    }
+}
+impl biome_diagnostics::Diagnostic for MigrationResults {
+    fn category(&self) -> Option<&'static biome_diagnostics::Category> {
+        Some(biome_diagnostics::category!("migrate"))
+    }
+
+    fn severity(&self) -> biome_diagnostics::Severity {
+        biome_diagnostics::Severity::Information
+    }
+
+    fn location(&self) -> biome_diagnostics::Location<'_> {
+        Location::builder()
+            .resource(&self.eslint_path.as_deref())
+            .build()
```

Comment:
cc @Conaclos here's a better way to construct the `Location`. We have a builder type for that :)